packages/flutter_tools/lib/src/template.dart

```diff
@@ -322,6 +322,18 @@ class Template {
 
       if (sourceFile.path.endsWith(templateExtension)) {
         final String templateContents = sourceFile.readAsStringSync();
+        final String? androidIdentifier = context['androidIdentifier'] as String?;
+        if (finalDestinationFile.path.endsWith('.kt') && androidIdentifier != null) {
+          final List<String> segments = androidIdentifier
+              .split('.')
+              .toList();
```

Comment:
I don't think toList() is required here. split() function returns a list.

Reply:
Sure, obviously a newbie dart developer :)

Reply:
Haha, no worries happens sometimes.